markhor_tracks/src/markhor_hw_interface.cpp

```diff
@@ -248,6 +252,19 @@ void MarkhorHWInterface::write()
 
 void MarkhorHWInterface::read(const ros::Duration& period)
 {
+
+  //{ "flipper_fl_motor_j", "flipper_fr_motor_j", "flipper_rl_motor_j", "flipper_rr_motor_j" };
+
+  joint_position_[0] = -2*3.14159265358*front_left_drive->GetSensorCollection().GetPulseWidthPosition() / (double)track_encoder_to_rad_coeff;
+  joint_position_[1] = 2*3.14159265358*front_right_drive->GetSensorCollection().GetPulseWidthPosition() / (double)track_encoder_to_rad_coeff;
+  joint_position_[2] = -2*3.14159265358*rear_left_drive->GetSensorCollection().GetPulseWidthPosition() / (double)track_encoder_to_rad_coeff;
+  joint_position_[3] = 2*3.14159265358*rear_right_drive->GetSensorCollection().GetPulseWidthPosition() / (double)track_encoder_to_rad_coeff;
+
+  joint_velocity_[0] = -2*3.14159265358*front_left_drive->GetSensorCollection().GetPulseWidthVelocity() / (double)track_encoder_to_rad_coeff;
+  joint_velocity_[1] = 2*3.14159265358*front_right_drive->GetSensorCollection().GetPulseWidthVelocity() / (double)track_encoder_to_rad_coeff;
+  joint_velocity_[2] = -2*3.14159265358*rear_left_drive->GetSensorCollection().GetPulseWidthVelocity() / (double)track_encoder_to_rad_coeff;
+  joint_velocity_[3] = 2*3.14159265358*rear_right_drive->GetSensorCollection().GetPulseWidthVelocity() / (double)track_encoder_to_rad_coeff;
+
   // Read from the motor API, going to read from the TalonSRX objects
   // ROS_INFO("Vel: %d, %d",
   // rear_right_drive->GetSensorCollection().GetQuadratureVelocity(),rear_left_drive->GetSensorCollection().GetQuadratureVelocity());
```

Comment:
I know its not part of your changes, but would it possible to remove this if applicable?

Reply:
yeah, I'll remove it